### PR TITLE
Sunsama task creation + cleanup [LOOK AT MERGE CONSTRAINTS]

### LINKED ISSUE #3: Refresh Expired Asana Tokens
Asana tokens expire after an hour but the application fails to handle it gracefully

### PR DESCRIPTION
**Pull request by zanemcca**
# MERGE CONSTRAINTS
Merge this after this [PR](https://github.com/zanemcca/task_planner_4000/pull/4)

## Completed
 - Add a command that allows for creating a sunsama task with time estimates and channels
   - Form UI/UX
   - tab through form fields
     - time estimate
     - channel
     - task date
 - Fixed token refresh [issue](https://github.com/zanemcca/task_planner_4000/issues/3)
 - Cleaned up the mobile UX/UI a bit (Still not ideal but it is functional)

## Preview
https://tp4000preview.zanemccaig.com/
https://www.loom.com/share/08d7627fe13c4d58a0cbca8910e69702

### CODE REVIEW

**Comment by zanemcca** [comment]
[Add Sunsama Task Creation Form](https://trello.com/c/UrwpkhuR/2-add-sunsama-task-creation-form)

**Comment by zanemcca** [comment]
[Implement Responsive Design](https://trello.com/c/KB64fvu9/5-implement-responsive-design)

**Comment by zanemcca** [comment]
[Fix token refresh issue](https://trello.com/c/d1of9X3E/6-fix-token-refresh-issue)

**Comment by priyadarshy** [comment]
Nice work here! I can create a Sunsama task for any given day without leaving my keyboard. It's pretty cool how seamless it is. This UI also gives me a good idea of how the command palette could be extended to quickly let customers create tasks with rich information in any of their services! 

I mentioned this in the Asana PR but I think I'd recommend the same two improvements here:
- Enter should get you to submit immediately so you don't need to tab through cancel
- Don't wait on completion (although in this case I noticed you are "simulating" the latency!)

Some other improvements I might recommend:
- Re-order the fields as such: title, channel, duration, and then date. For some reason, this feels more intuitive to me but I can't put my finger on why. 
- Replace the Duration field with a dropdown of common values a la the Sunsama web app
- I might even get rid of the cancel button and rely on the "X" in the top right corner and ESC

**Review by priyadarshy — APPROVED** [review]
Browsed through code, tested on preview branch (not locally).

**Comment by zanemcca** [comment]
- **"Don't wait on completion (although in this case I noticed you are "simulating" the latency!)"**
  - Yeah it felt weird with the new task immediately going to the list without any sort of animation. This was a quick and dirty fix that gives the user a visual queue that work was happening

- **"Re-order the fields as such: title, channel, duration, and then date. For some reason, this feels more intuitive to me but I can't put my finger on why."**
  - I can get behind this 👍 

- **"Replace the Duration field with a dropdown of common values a la the Sunsama web app"**
  - I totally should have picked up how you guys did that feature in Sunsama. It would have actually made the UI a little easier for me. The path I took to get to the current solution was a bit of a meander.

- **"I might even get rid of the cancel button and rely on the "X" in the top right corner and ESC"**
  - I really dig it. It gets the user one tab closer to the finish line without mistakenly cancelling their work.

**Review by BrunoBernardino — COMMENTED** [review]
Didn't pull down since Ashutosh already reviewed it, but glanced through the code, the changes look good!

File: src/components/Commands.tsx [hunk 1]
```diff
@@ -48,17 +45,130 @@ const Command = (props: any) => {
...
+        setIsAsanaTaskFormVisible(true)
+      }
+    }
+  }]
+
+  const handleAsanaTaskCreateCancel = () => {
+    const { form } = asanaFormRef.props;
+    form.resetFields();
+    setIsAsanaTaskFormVisible(false)
+  }
```
Comment (bone-saw) [inline src/components/Commands.tsx:83]: Today I learned. I didn't know you could access props from a child component like this. I would have just passed the `setIsAsanaTaskFormVisible` down to the child as a prop. I see the attraction in doing it this way instead.

**Pull request merged**